downloader.js

```diff
@@ -1,28 +1,30 @@
-const bluebird = require('bluebird')
-const fse = bluebird.promisifyAll(require('fs-extra'))
-const loc = require('./utils/location.js')
+const { startQueue } = require('./utils/downloadQueue.js')
+const BulkSender = require('./utils/BulkSender.js')
 
-function downloadChapter (event, args, worker) {
-  event.sender.send('recv-download-chapter', Object.assign({}, args, { err: null }))
-
-  // Send the download chapter message to the downloader process.
-  worker.send(args)
-}
-
-function deleteChapter (basePath, args) {
-  const path = loc.chapterPath(basePath, args.mangaName, args.chapterNum)
-
-  return fse.removeAsync(path)
+function downloadProcessHandler (path, file) {
+  const sender = new BulkSender((bulkMsg) => process.send(bulkMsg))
+  startQueue(path, file, (msg) => sender.add(msg)).then((queue) => {
```

Comment:
Haven't looked at the internal code for the async version of the queue yet. I might be able to use their implementation to run tasks in parallel though since right now all tasks are run sequentially.